### ISSUE TITLE
Proposal to make max_iter a parameter for the is_inside function inside Room class

### ISSUE THREAD

**Issue by grantgasser** [LCAV/pyroomacoustics#150]
Hi there.

We're working on a project where we are constantly moving the mic array around the room and have to use the `is_inside` function to check if we are still inside the room. Occasionally we get this error:

```Traceback (most recent call last):

  File "<path>", line 222, in step
    points = np.array([x, y]) if self.room.is_inside([x, y], include_borders=False) else self.agent_loc
  File "<path>", line 1491, in is_inside
    '''
ValueError:  
                Error could not determine if point is in or out in maximum number of iterations.
                This is most likely a bug, please report it.
```

I've looked at `is_inside` and there is a variable `max_iter` that is set to 5. When I increase that variable to something higher, we don't seem to get the above error (at least it would be much less likely). I'd like to propose making `max_iter=5` a parameter in the function where 5 is the default but the caller can choose a different value. 

I am willing to make this change and submit a PR. Wanted to get some input first. 

Thanks!

**Comment by fakufaku** [comment]
Hi @grantgasser , thanks for reaching out! That seems to be a problem indeed! However, I'm a little bit reluctant to make this a parameter as the implementation details should be transparent to the user.

My understanding is that the likelihood of the method to fail should decrease exponentially with the value of `max_iter`. However, it might still fail if you try hard enough. Thus the maximum value we should pick is also the longest time we are potentially willing to wait.

Here is what I propose to do:
* Add a package wide configuration in `pyroomacoustics/parameters.py`. This way the parameter is hidden from users, but can still be changed if it fails for some specific applications.
* Set the default value to something higher, say `max_iter=20`, or more. What are you using now ?

It'd be great if you make a PR! This would make the package more robust in general. Thank you very much!

**Comment by grantgasser** [comment]
Hi @fakufaku. Thanks for getting back to me! I used 20 and that seemed to work fine for what we were doing. Raised it to 50 just in case. I can set `max_iter=20` in `parameters.py`. 

Should I add it to the `_constants_default` ?

**Comment by fakufaku** [comment]
Sounds good. Yes, you can add it to `_constants_default`. Maybe we can name the variable `room_isinside_max_iter`. It is a mouthful, but we shouldn't have to use it all that often.

**Comment by grantgasser** [comment]
Solved with #151
